afterpay/src/main/kotlin/com/afterpay/android/view/AfterpayPriceBreakdown.kt

```diff
@@ -75,20 +87,74 @@ class AfterpayPriceBreakdown(context: Context, attrs: AttributeSet?) : FrameLayo
-                .append(
+            text = SpannableStringBuilder().apply {
+                if (instalment is AfterpayInstalment.NotAvailable) {
+                    append(" ", CenteredImageSpan(drawable), Spannable.SPAN_INCLUSIVE_EXCLUSIVE)
+                    append(" ")
+                    append(content.text)
+                } else {
+                    append(content.text)
+                    append(" ")
+                    append(" ", CenteredImageSpan(drawable), Spannable.SPAN_INCLUSIVE_EXCLUSIVE)
+                }
+                append(" ")
+                append(
                     resources.getString(R.string.price_breakdown_info_link),
                     URLSpan("https://static-us.afterpay.com/javascript/modal/us_modal.html"),
                     Spannable.SPAN_INCLUSIVE_EXCLUSIVE
                 )
-            contentDescription = resources.getString(R.string.price_breakdown_content_description)
+            }
+            contentDescription = content.description
         }
     }
+
+    private fun generateContent(afterpay: AfterpayInstalment): Content = when (afterpay) {
+        is AfterpayInstalment.Available ->
+            Content(
+                text = String.format(
+                    resources.getString(R.string.price_breakdown_total_cost),
+                    afterpay.instalmentAmount
+                ),
+                description = String.format(
+                    resources.getString(R.string.price_breakdown_total_cost_description),
+                    afterpay.instalmentAmount
+                )
+            )
+        is AfterpayInstalment.NotAvailable ->
+            if (afterpay.minimumAmount != null)
+                Content(
+                    text = String.format(
+                        resources.getString(R.string.price_breakdown_limit),
+                        afterpay.minimumAmount,
+                        afterpay.maximumAmount
+                    ),
+                    description = String.format(
+                        resources.getString(R.string.price_breakdown_limit_description),
+                        afterpay.minimumAmount,
+                        afterpay.maximumAmount
+                    )
+                )
+            else
+                Content(
+                    text = String.format(
+                        resources.getString(R.string.price_breakdown_upper_limit),
+                        afterpay.maximumAmount
+                    ),
+                    description = String.format(
+                        resources.getString(R.string.price_breakdown_upper_limit_description),
+                        afterpay.maximumAmount
+                    )
+                )
+        AfterpayInstalment.NoConfiguration ->
+            Content(
+                text = resources.getString(R.string.price_breakdown_no_configuration),
+                description = resources.getString(R.string.price_breakdown_no_configuration_description)
+            )
+    }
```

Comment:
This is potentially something that could live on the Instalment to be testable? Although do you need an android context to get resources?

Reply:
Yeah unfortunately the reliance on Android resources don't make it particularly easy to test.

We can absolutely test the `totalAmount` to `AfterpayInstalment` conversion and is something I'll do if I get the chance.